### DISCUSSION TITLE
speechlib discussion

### DISCUSSION THREAD

**Discussion by NavodPeiris** [NavodPeiris/speechlib#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by gety9** [comment]
@Navodplayer1 

Thanks for interesting project.

Could you please tell why speechlib can't recognize speakers here?
(i don't need names, but speaker 1 and speaker 2 will do :)
voice folder is provided, but empty...)

Update: now that i look at output, i see that speaker recognition took 0 seconds. I assumed if voice files not provided it will at least recognize speaker 1 / speaker 2. (without names = folder name of each speaker). Or am i wrong?

Code and files:
```
from speechlib import Transcriptor

file = "/content/drive/MyDrive/Colab Notebooks/PaypalCEOAlexChrissW.mp3"
voices_folder = "/content/drive/MyDrive/Colab Notebooks/voices"
language = "en"
log_folder = "/content/drive/MyDrive/Colab Notebooks/logs"
modelSize = "large"
quantization = False

transcriptor = Transcriptor(file, log_folder, language, modelSize, voices_folder, quantization)

res = transcriptor.transcribe()
```

**Comment by gety9** [comment]
![Screenshot_2024-01-22_16-11-53](https://github.com/Navodplayer1/speechlib/assets/16415882/83c725d4-d5c7-4b64-8a23-5813433b1611)

**Comment by gety9** [comment]
[output_2024-01-22.txt](https://github.com/Navodplayer1/speechlib/files/14015499/output_2024-01-22.txt)

**Comment by gety9** [comment]
Github doesn't allow to upload mp3 file... let me see if i can find a way :)

update: https://whyp.it/tracks/152176/paypalceoalexchrissw?token=qX9Th

**Comment by gety9** [comment]
By the way, i justed wanted to transcript interview in swedish, provided mp3 file, but forgot to change language (it was left as language = "en")

So speechlib did transcript AND translation :)
and seems accurate. That's amazing.

-----
Still no corect speaker 1, speaker 2 (all are "unknow")

**Comment by NavodPeiris** [comment]
@gety9 you can leave the voices_folder as empty string "" . then you will get SPEAKER_00, SPEAKER_01 ... etc. let me know if this solved your issue.

**Comment by lsnsk** [comment]
speechlib is great!
Can we use WeSpeaker onnx models for diarization? Now I need wespeaker/voxblink2_samresnet100_ft.onnx

**Reply by NavodPeiris** [reply]
curently we don't support onnx models
